OpenTrons/opentrons-api
# chore(app,app-shell,app-shell-odd, protocol-designer, labware-library, components, labware-designer): update Electron version to v27.0.0

# Overview
Update Electron version 27.0.0
https://www.electronjs.org/docs/latest/tutorial/electron-timelines#timeline

There is `loadURL` issue in the versions that include the following PR.
https://github.com/electron/electron/pull/40143

At this moment we cannot update any newer version. (I'll keep checking the repo)


This PR also includes Nodejs update because of Electron v27's requirement.
This PR needs Nodejs v21.4.0 since M1 Mac has an issue about running Desktop app. (I'll keep an eye on this to support more nodejs version). M2 Mac can use nodejs v18.

Need `NODE_OPTION` due to OpenSSL version.
I tried to export NODE_OPTION in the top of Makefile, but somehow that has an impact on other part and prevent building the app(Desktop/ODD). I use `NODE_OPTION` separately.
Another reasons for this are `labware-library` and `protocol-designer` since the option prevents Cypress.


### Running the application
node v21
- [x] ODD mode
- [x] ODD

### CI
- [x] update Nodejs version (v16 -> v21)

### OT-2
- [x] USB connection

### Other applications
- [x] Protocol Designer
- [x] Labware Library
- [x] Labware Designer

### Commands
- [x] make check-js
- [x] make lint-js
- [x] make test-js

This PR removes `usb-detection` since the repo has been archived and not compatible with node v18+.
The new package we need to use is `[node-usb](https://github.com/node-usb/node-usb)`
https://working-wormhole-d95.notion.site/usb-detection-node-usb-a6637f72f5d44f3ba9c7520b4f0da24f



# Test Plan
- Run Desktop app and check that the app works as well as the current edge
- Run ODD app and check that the app works as well as the current edge


# Changelog



# Review requests



# Risk assessment
